### ISSUE TITLE
AI reluctant to build early eras buildings

### ISSUE THREAD

**Issue by Infixo** [LoneGazebo/Community-Patch-DLL#3802]
_1. Mod version (i.e Date - 4/23): 12-15
_2. Mod list (if using Vox Populi only, leave blank): none
_3. Error description:
I prepared few tweaks for AI building selection process.
- Buildings from previous eras are valued more (actually, current algorithm doesn't differenciate at all, seems like a bug rather)
- Tweaks to Defensive buildings, to account for crime unhapiness, also little nerf to pure Defense ones that were build always as first priority; Walls and Castles are still getting to the top, but Arsenal is sometimes postponed in favor for other buildings
- Parameters that are used to recalculate value depending on cost (turns) tweaked a little, so low-cost buildings (1-3 turns) are bumped over high-cost (8+ turns) ones.
- Tweaks to happiness-related buildings, not changing the output but placing values into more comparable ranges
- TODO: Customs House still gets scores like 500000+ or 700000+ and is unbeatable in many cases. Haven't found why yet.

I attach files:
- zip contains 2 CityStrategyAIProductionLogs, one from original 12-15 and the other after the changes; 10 turns total
- text file is an extract with some examples of results
[buildings.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/1602244/buildings.zip)
[ex2.txt](https://github.com/LoneGazebo/Community-Patch-DLL/files/1602245/ex2.txt)

TL;DR Puppet cities build Monuments, Walls, Barracks, Libraries, etc. all missing early infrastructure. Well-developed cities do more or less the same as before, but older buildings appear more often, also cheap support buildings like Garden, Forge, Ligthouse are chosen more often.
